tests/base/src/test/java/org/wildfly/security/x500/X509PeerCertificateChainEvidenceDecoderTest.java

```diff
@@ -278,4 +300,24 @@ private void addUser(ModifiableSecurityRealm realm, String userName, String role
         realmIdentity.setAttributes(attributes);
         realmIdentity.dispose();
     }
+
+    /**
+     * Custom Realm Mapper which verifies evidence is not null and maps to default realm.
+     * ONLY for testing.
+     */
+    private class CustomRealmMapper implements RealmMapper {
+
+        public CustomRealmMapper() {
+        }
+
+        @Override
+        public String getRealmMapping(Principal principal, Evidence evidence) {
+            if (principal instanceof NamePrincipal) {
```

Comment:
In this test, the `principal` will be an `X500Principal` instead of a `NamePrincipal` so I don't think the test is actually entering this block. Perhaps your security domain could have two security realms defined and then your realm mapper could assign a realm depending on whether or not `evidence` is null and then the test case could assert that `sac.authorize()` is true (to make sure the correct realm was selected). The test should fail before your changes and pass with your changes.

Reply:
Thanks for the feedback! I created a new class with an updated test.